common/src/main/java/com/revenuecat/purchases/common/logWrapper.kt

```diff
@@ -0,0 +1,51 @@
+package com.revenuecat.purchases.common
+
+import com.revenuecat.purchases.strings.Emojis
+
+fun log(type: LogIntent, level: LogLevel, message: String) {
+    val emojifiedMessage = "${type.emojiList.joinToString("")} $message"
+    when (level) {
+        LogLevel.DEBUG -> debugLog(emojifiedMessage)
+        LogLevel.ERROR -> errorLog(emojifiedMessage)
+        LogLevel.INFO -> infoLog(emojifiedMessage)
+        LogLevel.WARN -> warnLog(emojifiedMessage)
+    }
```

Comment:
I like this LogIntent thingy, it cleaned up the previous logic a lot :)

Reply:
+1

Reply:
I ❤️  it, and it's pretty easy to expand upon. 
@tiinanguyen is the idea to eventually replace `infoLog(...` calls with `log(LogIntent.INFO, ...`? 
I think that's the ideal case for consistency. 
Doesn't have to be done in this PR, though.

Reply:
@aboedo Yup! `infoLog(...` will be replaced once I get to those in the string files.